__tests__/api/api.test.js

```diff
@@ -230,14 +230,85 @@ describe("GET /api/articles/:article_id/comments", () => {
   });
 });
 
-describe("ALL /notapath", () => {
-  test("404: responds with a status of 404 and a custom message when the path is not found", () => {
+describe("POST /api/articles/:article_id/comments", () => {
+  test("201: responds with a status of 201 and the comment object that has been sent", () => {
+    const newComment = {
+      body: "The answer is doughnuts",
+      username: "butter_bridge"
+    };
     return request(app)
-      .get("/api/banana")
-      .expect(404)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(201)
+      .then((response) => {
+        const { comment } = response.body;
+        expect(comment).toEqual({
+          article_id: 1,
+          author: "butter_bridge",
+          body: "The answer is doughnuts",
+          ...comment
+        });
+      });
+  });
+  test("400: Should return Bad request when the comment has a malformed body", () => {
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send({})
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("Bad request");
+      });
+  });
+  test("400: Should return 'Username Not Found' when the comment is missing username field", () => {
+    const newComment = {
+      body: "The answer is doughnuts"
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
       .then(({ body }) => {
         const { msg } = body;
-        expect(msg).toBe("Not found");
+        expect(msg).toBe("Username Not Found");
       });
   });
+  test("400: Should return 'Comment Not Found' when the comment is missing comment field", () => {
+    const newComment = {
+      username: "butter_bridge"
+    };
+    return request(app)
+      .post("/api/articles/1/comments")
+      .send(newComment)
+      .expect(400)
+      .then(({ body }) => {
+        const { msg } = body;
+        expect(msg).toBe("Comment Not Found");
+      });
+  });
+  test("400: Should return 'Bad request' when the article_id is banana", () => {
```

Comment:
Be specific about why banana is bad, something like, when the article id is invalid